.github/workflows/ci.yml

```diff
@@ -2,12 +2,12 @@ name: Python CI
 
 on:
   push:
-    branches: [master]
+    branches: [ master ]
   pull_request:
-    branches: [master]
+    branches: [ master ]
 
 concurrency:
-  group: ${{ format('ci-{0}', format('pr-{0}', github.event.pull_request.number) || github.sha) }}
+  group: ${{ format('ci-{0}', github.head_ref && format('pr-{0}', github.event.pull_request.number) || github.sha) }}
   cancel-in-progress: ${{ github.event_name == 'pull_request' }}
 
 jobs:
```

Comment:
Matrix all builds and tests across windows, macOS and Linux. Had another bug sneak through

Reply:
#3133